addons/website_slides/models/slide_slide.py

```diff
@@ -877,16 +899,36 @@ def _action_set_completed(self, target_partner):
             'vote': 0,
             'completed': True} for new_slide in new_slides])
 
-        return True
+    def action_set_uncompleted(self):
+        if any(not slide.can_be_undone for slide in self):
+            raise UserError(_('You cannot mark a slide as uncompleted if you are not among its members.'))
+
+        # Remove the Karma point gained
+        self._action_set_quiz_done(completed=False)
 
-    def _action_set_quiz_done(self):
+        self.env['slide.slide.partner'].sudo().search([
+            ('slide_id', 'in', self.ids),
+            ('partner_id', '=', self.env.user.partner_id.id)
+        ]).completed = False
+
+    def _action_set_quiz_done(self, completed=True):
+        """Add or remove karma point related to the quiz.
+
+        :param completed: True if the quiz will be marked as completed (karma will be increased)
+        """
         if any(not slide.channel_id.is_member for slide in self):
```

Comment:
Should probably check published state for consistency with the "can_marl_complete" stuff.